src/features/player-history-aggregator/services/player.history.service.js

```diff
@@ -1,23 +1,80 @@
 import { JSDOM } from "jsdom";
+import { TrackedPlayers } from "../../../models/tracked.players.js";
 import { Players } from "../../../models/players.js";
 
 export function addCurrentPlayersFromSteam(players, games) {
   return games.map((game, i) => {
-    game.playerHistory.push(new Players(players[i]));
+    let existingMonthAndYearIndex = getExistingMonthAndYearIndex(game);
```

Comment:
This logic belongs into the game and player classes. You need a method like `game.addOnePlayerHistoryEntry(players[i])` which takes care of updating the `playerHistory` property of the game class.

More details below.